Common/src/linear_algebra/CSysSolve.cpp

```diff
@@ -241,7 +242,7 @@ unsigned long CSysSolve<ScalarType>::CG_LinSolver(const CSysVector<ScalarType> &
 
     norm_r = r.norm();
     norm0  = b.norm();
-    if ( (norm_r < tol*norm0) || (norm_r < eps) ) {
+    if ( ((norm_r < tol*norm0) || (norm_r < eps)) && !TapeActive ) {
```

Comment:
Are you recording the linear solver iterations to have to do this? :S

Reply:
No, by no means. If we recording though, we don't want to return at this point in case the primal solution is fully converged, but we still want the solution update being included.
It's a fix that's not multizone-specific.

Reply:
It is in CSysSolve::Solve.
The legacy grid movement calls the individual methods directly but that does not matter because it does the adjoint analytically (symmetric matrix).

Reply:
Alright.